concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs

```diff
@@ -590,6 +596,57 @@ deriving via (LiftMonadBlobStore (ExceptT e) m)
 deriving via (LiftMonadBlobStore (ReaderT r) m)
     instance MonadBlobStore m => MonadBlobStore (ReaderT r m)
 
+-- * In-memory blob store
+
+-- |An in-memory blob store. This is intended for testing purposes, as it can be more convenient
+-- than working with files. Internally, the blob store is handled as a lazy 'LBS.ByteString' guarded
+-- by an 'MVar'.
+newtype MemBlobStore = MemBlobStore {theMemBlobStore :: MVar LBS.ByteString}
+
+-- |Create a fresh, empty 'MemBlobStore'.
+newMemBlobStore :: IO MemBlobStore
+newMemBlobStore = MemBlobStore <$> newMVar LBS.empty
+
+-- |A monad transformer that provides an instance of 'MonadBlobStore' based on a 'MemBlobStore'.
+-- Note, 'getCallbacks' is currently not implemented.
+newtype MemBlobStoreT m a = MemBlobStoreT {runMemBlobStoreT :: MemBlobStore -> m a}
+    deriving
+        (Functor, Applicative, Monad, MonadReader MemBlobStore, MonadIO, MonadFail, MonadLogger, MonadCatch.MonadThrow, MonadCatch.MonadCatch)
+        via (ReaderT MemBlobStore m)
+    deriving
+        (MonadTrans)
+        via (ReaderT MemBlobStore)
+
+instance MonadIO m => MonadBlobStore (MemBlobStoreT m) where
+    storeRaw b = do
+        mv <- asks theMemBlobStore
+        liftIO $ do
+            bs <- takeMVar mv
+            let !offset = LBS.length bs
+            let !newBS = Builder.toLazyByteString $
+                    Builder.lazyByteString bs
+                    <> Builder.word64BE (fromIntegral $ BS.length b)
+                    <> Builder.byteString b
+            putMVar mv newBS
+            return (BlobRef $ fromIntegral offset)
+    loadRaw (BlobRef offset) = do
+        mv <- asks theMemBlobStore
+        bs <- liftIO (readMVar mv)
+        let bs' = LBS.drop (fromIntegral offset) bs
+        case runGetLazy getWord64be (LBS.take 8 bs') of
+            Left e -> error e
+            Right len -> return $! LBS.toStrict (LBS.take (fromIntegral len) (LBS.drop 8 bs'))
+    flushStore = return ()
+
+    getCallbacks = error "getCallbacks not supported for MemBlobStore"
```

Comment:
I don't see that the new test really crucially needs this memblobstore.

So while I think it would be nice to have, and  other tests should over time be migrated to it, we need to do this properly and implement the callbacks as well.

We also should add a test to make sure that the memblobstore and the actual blob store behave the same, so that the testing with mem blob store is trustworthy.